src/common/i18n.ts

```diff
@@ -62,48 +77,130 @@ let currentTranslations: Record<Locale, typeof Localizations[Locale]>;
 const i18n = {
     get locale() {return currentLocale;},
     get rules() {return ruleCache.get(i18n.locale);},
-    get supportedLocales() {return Object.keys(currentTranslations);},
-    isSupported(abbr: Locale) {return Object.keys(currentTranslations).includes(abbr);},
+    get supportedLocales() {return currentTranslations ? Object.keys(currentTranslations) : [];},
+    isSupported(abbr: Locale) {return currentTranslations ? Object.keys(currentTranslations).includes(abbr) : false;},
 
     // This can easily be made more generic if needed in the future
     init(options: InitOptions) {
         const {locale, fallback = "en-US", translations} = options;
+        if (!isValidLocale(fallback)) throw new Error(`Invalid fallback locale: ${fallback}`);
         currentFallback = fallback;
         currentTranslations = translations;
-        i18n.setLocale(locale);
+        i18n.setLocale(locale ?? fallback);
     },
 
-    setLocale(locale?: Locale) {
-        currentLocale = locale ?? currentFallback;
+    setLocale(locale: Locale) {
+        if (!i18n.isSupported(locale)) {
+            Logger.warn("i18n", `Locale ${locale} is not supported, falling back to ${currentFallback}`);
+            locale = currentFallback;
+        }
+
+        currentLocale = locale;
         if (ruleCache.has(currentLocale)) return;
         ruleCache.set(currentLocale, new Intl.PluralRules(currentLocale));
     },
 
-    pluralize(target: string | Plural, count: number) {
+    /** @private */
+    pluralize(target: string | Plural, count?: number): string {

```

Comment:
No need for jsdoc when we can use access modifiers
```suggestion
    private pluralize(target: string | Plural, count?: number): string {
```